Assets/HoloToolkit/Input/Tests/Scripts/SphereKeywords.cs

```diff
@@ -1,26 +1,36 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
 using UnityEngine;
 
 namespace HoloToolkit.Unity.InputModule.Tests
 {
-    public class SphereKeywords : MonoBehaviour
+    public class SphereKeywords : MonoBehaviour, ISpeechHandler
     {
         public void ChangeColor(string color)
         {
+            Material material = GetComponent<Renderer>().material;
+
             switch (color.ToLower())
             {
                 case "red":
-                    GetComponent<Renderer>().material.color = Color.red;
+                    material.color = Color.red;
                     break;
                 case "blue":
-                    GetComponent<Renderer>().material.color = Color.blue;
+                    material.color = Color.blue;
                     break;
                 case "green":
-                    GetComponent<Renderer>().material.color = Color.green;
+                    material.color = Color.green;
                     break;
             }
+
+            Resources.UnloadAsset(material);
```

Comment:
This should be ` Destroy(_cachedMaterial);` and called in OnDestroy or when finished using the `_cachedMaterial`.